dbms/programs/client/Client.cpp

```diff
@@ -327,6 +327,78 @@ class Client : public Poco::Util::Application
             || (now.month() == 1 && now.day() <= 5);
     }
 
+    bool isChineseNewYearMode(const String & local_tz)
+    {
+        /// Days of Dec. 20 in Chinese calendar starting from year 2019 to year 2105
+        static constexpr UInt16 chineseNewYearIndicators[]
+            = {18275, 18659, 19014, 19368, 19752, 20107, 20491, 20845, 21199, 21583, 21937, 22292, 22676, 23030, 23414, 23768, 24122, 24506,
+               24860, 25215, 25599, 25954, 26308, 26692, 27046, 27430, 27784, 28138, 28522, 28877, 29232, 29616, 29970, 30354, 30708, 31062,
+               31446, 31800, 32155, 32539, 32894, 33248, 33632, 33986, 34369, 34724, 35078, 35462, 35817, 36171, 36555, 36909, 37293, 37647,
+               38002, 38386, 38740, 39095, 39479, 39833, 40187, 40571, 40925, 41309, 41664, 42018, 42402, 42757, 43111, 43495, 43849, 44233,
+               44587, 44942, 45326, 45680, 46035, 46418, 46772, 47126, 47510, 47865, 48249, 48604, 48958, 49342};
+        static constexpr size_t N = sizeof(chineseNewYearIndicators) / sizeof(chineseNewYearIndicators[0]);
+
+        /// All time zone names are acquired from https://www.iana.org/time-zones
+        static constexpr const char * chineseNewYearTimeZoneIndicators[] = {
+            /// Time zones celebrating Chinese new year.
+            "Asia/Shanghai",
```

Comment:
BeiJing?

Reply:
There isn't one in iana's tz database and the reason is described there

Reply:
The time zone of the region is named by the biggest city of that region at the time of introduction of time zone. So, Beijing is using Asia/Shanghai.